cli/parse_test.ts

```diff
@@ -0,0 +1,725 @@
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("fn() is called for alias option", () => {
+    const expected = { options: { foo: "changed" }, arguments: {} };
+    const actual = parse(["-f=bar"], {
+      options: [
+        {
+          name: "foo",
+          alias: "f",
+          type: String,
+          value: { name: "VALUE" },
+          fn: () => "changed",
+        },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+
+  await t.step("string type value", () => {
+    const expected = { options: { foo: "1" }, arguments: {} };
+    const actual = parse(["--foo=1"], {
+      options: [
+        { name: "foo", type: String, value: { name: "VALUE" } },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("throws on type string with non-string default value", () => {
+    assertThrows(() => {
+      parse(["--foo"], {
+        options: [
+          { name: "foo", type: String, default: true as unknown as string },
+        ],
+      });
+    });
+  });
+
+  await t.step("number type value", () => {
+    const expected = { options: { foo: 1 }, arguments: {} };
+    const actual = parse(["--foo=1"], {
+      options: [
+        { name: "foo", type: Number, value: { name: "VALUE" } },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("throws on type number with non-number value", () => {
+    assertThrows(() => {
+      parse(["--foo=bar"], {
+        options: [
+          { name: "foo", type: Number, value: { name: "VALUE" } },
+        ],
+      });
+    });
+  });
+  await t.step("throws on type number with non-number default value", () => {
+    assertThrows(() => {
+      parse(["--foo"], {
+        options: [
+          { name: "foo", type: Number, default: "fail" as unknown as number },
+        ],
+      });
+    });
+  });
+
+  await t.step("boolean type value", () => {
+    const expected = { options: { foo: true }, arguments: {} };
+    const actual = parse(["--foo=true"], {
+      options: [
+        { name: "foo", type: Boolean, value: { name: "VALUE" } },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("throws on type boolean with non-boolean value", () => {
+    assertThrows(() => {
+      parse(["--foo=bar"], {
+        options: [
+          { name: "foo", type: Boolean, value: { name: "VALUE" } },
+        ],
+      });
+    });
+  });
+  await t.step("throws on type boolean with non-boolean default value", () => {
+    assertThrows(() => {
+      parse(["--foo"], {
+        options: [
+          { name: "foo", type: Boolean, default: "fail" as unknown as boolean },
+        ],
+      });
+    });
+  });
+
+  await t.step("strip single quoted value", () => {
+    const expected = { options: { foo: "bar" }, arguments: {} };
+    const actual = parse(["--foo", "'bar'"], {
+      options: [
+        { name: "foo", value: { name: "VALUE" } },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
```

Comment:
Why? Bash will already remove the quotes. You can just leave them if someone passes them anyway by escaping them.

It reminds me of PHP automatically escaping and unescaping random strings without my consent.

Reply:
Again, fair point.